File: testdata/piper.py
#!/usr/bin/env python
from email.Parser import Parser
import sys

msg = sys.stdin.read()

p = Parser()
emailMessage = p.parsestr(msg)
headers = p.parsestr(msg, True)

print headers['message-id']
print headers['from']
print headers['in-reply-to']
print headers['references']
print headers['subject']

File: bin/piper.pl
#!/usr/bin/perl -w
use strict;
use Email::Simple;

my $message = '';
while(<>) {
    $message .= $_;
}


my $obj = Email::Simple->new($message);
print $obj->header("Message-ID")."\n";
print $obj->header("From")."\n";
print $obj->header("In-Reply-To")."\n";
print $obj->header("References")."\n";
print $obj->header("Subject")."\n";
